src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php

```diff
@@ -547,7 +546,7 @@ private function doRead($sessionId)
             $sessionRows = $selectStmt->fetchAll(\PDO::FETCH_NUM);
 
             if ($sessionRows) {
-                if ($sessionRows[0][1] < time()) {
+                if ($sessionRows[0][1] < time() || (isset($sessionRows[0][2]) && $sessionRows[0][1] + $sessionRows[0][2] < time())) {
```

Comment:
@nicolas-grekas This addition is actually never called, in BC mode or the new mode. But added regardless per https://github.com/symfony/symfony/pull/21423#discussion_r100334801.